utils/notificationhubServicePartition.ts

```diff
@@ -29,7 +35,13 @@ export function getNHService(
   throw new Error("It should not be called");
 }
 
-export function createNH0Service(): ExtendedNotificationHubService {
+/**
+ * @param config The NotificationHubConfig
+ * @returns a NotificationHubService used to call Notification Hub APIs
+ */
+export function createNHService(
+  config: NotificationHubConfig
```

Comment:
[Naming thing is hard](https://www.mediawiki.org/wiki/Naming_things), sometimes destructuring can help: 

```suggestion
  { AZURE_NH_HUB_NAME,  AZURE_NH_ENDPOINT }: NotificationHubConfig
```

Benefits:
* you have an explicit declaration of which property of an object will be used, right in the function signature
* you don't have the burden of choose a name for the object itself